src/jobs/rates-notification.job.ts

```diff
@@ -1,26 +1,27 @@
-import { Currency } from '../models/currency';
+import { SubscriptionsRepository } from '../repositories/subscriprion.repository';
 import { EmailService } from '../services/email.service';
-import { Privat24Client } from '../services/exchangers/privat24-client';
 import { ExchangerService } from '../services/exchanger.service';
+import logger from '../services/logger.service';
 import { SubscriptionsService } from '../services/subscription.service';
+import ukrainianBankExchangeHandler from '../services/bank-exchange-handler';
 
 export async function sendDailyRateEmail() {
   try {
-    const subscriptionService = new SubscriptionsService();
+    const subscriptionService = new SubscriptionsService(new SubscriptionsRepository());
     const emailService = new EmailService();
 
-    console.log(`[${new Date()}] going to send emails to subsribed users`);
-    const currentRate = await new ExchangerService(new Privat24Client()).getCurrentRate(Currency.USD);
+    logger.info(`[Scheduled job] [${new Date()}] going to send emails to subsribed users`);
+    const currentRate = await new ExchangerService(ukrainianBankExchangeHandler).getCurrentRate();
     const subcriptions = await subscriptionService.getAll();
     const emailAddresses = subcriptions.map((subscr) => subscr.email);
     for (const emailAddress of emailAddresses) {
       try {
         await emailService.sendCurrencyRateEmail({ to: emailAddress, currencyRate: currentRate });
       } catch (error) {
-        console.error('Error sending currency exchange rate: ', error);
+        logger.error(`Error sending currency exchange rate: ${JSON.stringify(error)}`);
       }
     }
   } catch (error) {
```

Comment:
here I might be a small optimisation, instead of awaiting each promise we might build an array of promises and then await all of them using Promise.allSettled, so the email will be sent concurrently

Reply:
agreed